src/lib/features/feature-lifecycle/feature-lifecycle-service.ts

```diff
@@ -95,13 +96,20 @@ export class FeatureLifecycleService extends EventEmitter {
         });
         this.eventBus.on(
             CLIENT_METRICS,
-            async (event: ValidatedClientMetrics) => {
-                if (event.environment) {
-                    const features = Object.keys(event.bucket.toggles);
-                    const environment = event.environment;
-                    await this.checkEnabled(() =>
-                        this.featuresReceivedMetrics(features, environment),
-                    );
+            async (events: IClientMetricsEnv[]) => {
+                if (events.length > 0) {
+                    const groupedByEnvironment = groupBy(events, 'environment');
+
+                    for (const [environment, metrics] of Object.entries(
+                        groupedByEnvironment,
+                    )) {
+                        const features = metrics.map(
+                            (metric) => metric.featureName,
+                        );
+                        await this.checkEnabled(() =>
```

Comment:
I'm assuming it's not gonna be more than a few envs. Because if the loop has 100s of entries then sequential awaits would kill performance

Reply:
I am not sure how edge is sending metrics, I am not sure it is env by env. So just in case I made the grouping to support more than one env and not assume that every item has same env.